src/mapped-types.ts

```diff
@@ -554,3 +554,23 @@ type IfEquals<X, Y, A = X, B = never> = (<T>() => T extends X
  *   gross(eur);
  */
 export type Brand<T, U> = T & { __brand: U };
+
+/**
+ * Optional
+ * @desc From `T` make a set of properties by key `K` become optional
+ * @example
+ *    type Props = {
+ *      name: string;
+ *      age: string;
+ *      height: number;
+ *    };
+ *
+ *    // Expect: { name?: string; age?: string; height?: number; }
+ *    type Props = Optional<Props>;
+ *
+ *    // Expect: { name: string; age?: string; height?: number; }
+ *    type Props = Optional<Props, 'age' | 'height'>;
+ */
+export type Optional<T extends {}, K = keyof any> = K extends (keyof T)
```

Comment:
Please change `{}` to `object` as this is a convention used in this project

Reply:
done